File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/repository/LibrarianRepository.java
package ru.spring5.elena.librarymanager.domain.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import ru.spring5.elena.librarymanager.domain.entity.LibrarianEntity;

@Repository
@Transactional(propagation = Propagation.MANDATORY)
public interface LibrarianRepository extends JpaRepository<LibrarianEntity, Long> {

}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/domain/repository/LibrarianRepository.kt
package ru.elenacherev.librarymanager.domain.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import ru.elenacherev.librarymanager.domain.entity.LibrarianEntity
import java.util.*

@Repository
@Transactional(propagation = Propagation.MANDATORY)
interface LibrarianRepository : JpaRepository<LibrarianEntity, UUID>